net/redisclient.go

```diff
@@ -226,11 +249,43 @@ func NewRedisRingClient(ro *RedisOptions) *RedisRingClient {
 		r.ring = redis.NewRing(ringOptions)
 		r.log = ro.Log
 		r.metricsPrefix = ro.MetricsPrefix
+
+		if ro.AddrUpdater != nil {
+			if ro.UpdateInterval == 0 {
+				ro.UpdateInterval = defaultUpdateInterval
+			}
+			go func() {
+				old := len(ringOptions.Addrs)
+				t := time.NewTicker(ro.UpdateInterval)
+				r.log.Info("Start goroutine to update redis instances every %s", ro.UpdateInterval)
+				for range t.C {
+					select {
+					case <-r.quit:
+						return
+					default:
+					}
+					addrs := ro.AddrUpdater()
+					if old != len(addrs) && len(addrs) != 0 {
+						r.log.Infof("Redis updater updating %d != %d", old, len(addrs))
+						r.SetAddrs(context.Background(), addrs)
+					}
+				}
+			}()
+		}
 	}
 
 	return r
 }
 
+func createAddressMap(addrs []string) map[string]string {
+	res := make(map[string]string)
+	for idx, addr := range addrs {
+		addr = strings.TrimPrefix(addr, "TCP://")
+		res[fmt.Sprintf("redis%d", idx)] = addr
```

Comment:
Is it safe to use index as a key? What if there two addresses `[10.2.0.1, 10.2.0.2]` just change order to `[10.2.0.2, 10.2.0.1]`? Myybe we should use address as a key here `res[addr] = addr`

Reply:
Yes you are right we could do that.
fixed